gclient/src/node/mod.rs

```diff
@@ -36,16 +36,45 @@ pub struct Node {
 }
 
 impl Node {
-    /// Instantiates a node runnin in development mode (--dev) via spawning its
-    /// process using the specified `path` on a randomly picked port. Waits for
-    /// the node getting initialized before returning it to a caller.
-    /// Ideally, the node's binary should be downloaded by means of CI pipeline
-    /// from https://get.gear.rs./
+    /// Instantiates a Gear node running in development mode (`--dev`) via
+    /// spawning its process using the specified `path` on a randomly picked
+    /// port. Waits for the node getting initialized before returning it to
+    /// a caller. Ideally, the node's binary should be downloaded by means
+    /// of CI pipeline from <https://get.gear.rs>.
     pub fn try_from_path(path: impl AsRef<OsStr>) -> Result<Self> {
+        Self::try_node_from_path(path, vec!["--dev"])
+    }
+
+    /// Instantiates a Vara node running in development mode (`--dev`) via
+    /// spawning its process using the specified `path` on a randomly picked
+    /// port. Waits for the node getting initialized before returning it to
+    /// a caller. Ideally, the node's binary should be downloaded by means
+    /// of CI pipeline from <https://get.gear.rs>.
+    pub fn try_vara_from_path(path: impl AsRef<OsStr>) -> Result<Self> {
+        Self::try_node_from_path(
+            path,
+            vec![
+                "--chain=vara-dev",
+                "--force-authoring",
+                "--rpc-cors=all",
+                "--alice",
+                "--tmp",
+            ],
+        )
+    }
+
+    /// Returns Web Socket address the node is listening to.
+    pub fn ws_address(&self) -> &WSAddress {
+        &self.ws_address
+    }
+
+    fn try_node_from_path(path: impl AsRef<OsStr>, args: Vec<&str>) -> Result<Self> {
         let port = port::pick();
         let port_string = port.to_string();
 
-        let args = vec!["--ws-port", &port_string, "--tmp", "--dev"];
+        let mut args = args;
```

Comment:
looks like there is no need for mutable rebinding. Argument can be marked `mut` directly in paratemeters list

Reply:
I don't like `mut` in the function signature when it is not required there by the function contract as it looks a little bit confusing in my opinion.

Reply:
But after reading this comment I've changed my mind: https://stackoverflow.com/questions/31388644/whats-the-idiomatic-way-to-pass-by-mutable-value#comment50755158_31388745

Reply:
But I get another error 🙈 

```
error[E0597]: `port_string` does not live long enough
  --> gclient/src/node/mod.rs:77:19
   |
71 |     fn try_node_from_path(path: impl AsRef<OsStr>, mut args: Vec<&str>) -> Result<Self> {
   |                                                                  - let's call the lifetime of this reference `'1`
...
77 |         args.push(&port_string);
   |         ----------^^^^^^^^^^^^-
   |         |         |
   |         |         borrowed value does not live long enough
   |         argument requires that `port_string` is borrowed for `'1`
...
93 |     }
   |     - `port_string` dropped here while still borrowed
```

So let's keep it as is.